File: app/src/main/java/com/credenceid/sample/epassport/LaunchActivity.java
package com.credenceid.sample.epassport;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

import com.credenceid.biometrics.Biometrics;
import com.credenceid.biometrics.BiometricsManager;
import com.credenceid.biometrics.DeviceFamily;
import com.credenceid.biometrics.DeviceType;

import static android.widget.Toast.LENGTH_LONG;
import static android.widget.Toast.LENGTH_SHORT;
import static com.credenceid.biometrics.Biometrics.ResultCode.OK;

public class LaunchActivity
		extends Activity {

	private static final String TAG = LaunchActivity.class.getSimpleName();

	/* CredenceSDK biometrics object, used to interface with APIs. */
	@SuppressLint("StaticFieldLeak")
	private static BiometricsManager mBiometricsManager;
	/* Stores which Credence family of device's this app is running on. */
	private static DeviceFamily mDeviceFamily = DeviceFamily.InvalidDevice;
	/* Stores which specific device this app is running on. */
	private static DeviceType mDeviceType = DeviceType.InvalidDevice;

	public static BiometricsManager
	getBiometricsManager() {
		return mBiometricsManager;
	}

	@SuppressWarnings("unused")
	public static DeviceFamily
	getDeviceFamily() {
		return mDeviceFamily;
	}

	@SuppressWarnings("unused")
	public static DeviceType
	getDeviceType() {
		return mDeviceType;
	}

	@Override
	protected void
	onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		/*  Create new biometrics object. */
		mBiometricsManager = new BiometricsManager(this);

		/* Initialize object, meaning tell CredenceService to bind to this application. */
		mBiometricsManager.initializeBiometrics((Biometrics.ResultCode resultCode,
												 String minimumVersion,
												 String currentVersion) -> {
			if (resultCode == OK) {
				Toast.makeText(this, "Biometrics initialized.", LENGTH_SHORT).show();

				mDeviceFamily = mBiometricsManager.getDeviceFamily();
				mDeviceType = mBiometricsManager.getDeviceType();

				/* Launch main activity. */
				Intent intent = new Intent(this, MRZActivity.class);
				intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
				startActivity(intent);
			} else Toast.makeText(this, "Biometrics FAILED to initialize.", LENGTH_LONG).show();
		});
	}
}

File: app/src/main/java/com/credenceid/sample/epassport/LaunchActivity.kt
package com.credenceid.sample.epassport

import android.Manifest
import android.app.Activity
import android.content.Intent
import android.content.pm.PackageManager
import android.os.Bundle
import android.widget.Toast
import android.widget.Toast.LENGTH_LONG
import android.widget.Toast.LENGTH_SHORT
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import com.credenceid.biometrics.Biometrics
import com.credenceid.biometrics.Biometrics.ResultCode.*
import com.credenceid.biometrics.BiometricsManager
import com.credenceid.biometrics.DeviceFamily
import com.credenceid.sample.epassport.modules.useCaseModule
import com.credenceid.sample.epassport.modules.viewModelModule
import com.google.android.material.snackbar.Snackbar
import org.koin.android.ext.koin.androidContext
import org.koin.core.context.startKoin
import timber.log.Timber

class LaunchActivity : Activity() {


    companion object {
        private const val PERMISSION_REQUEST_CODE = 123
    }

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)

        Timber.plant(Timber.DebugTree())

        startKoin {
            // declare used Android context
            androidContext(this@LaunchActivity)
            modules(viewModelModule)
            modules(useCaseModule)
        }

        // Initialize a new instance of ManagePermissions class
        checkPermissions()
    }

    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<out String>,
        grantResults: IntArray
    ) {
        if (requestCode == PERMISSION_REQUEST_CODE) {
            this.initBiometrics()
        }
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
    }

    private fun initBiometrics() {

        /*  Create new biometrics object. */
        App.BioManager = BiometricsManager(this)

        /* Initialize object, meaning tell CredenceService to bind to this application. */
        App.BioManager!!.initializeBiometrics { rc: Biometrics.ResultCode,
                                                _: String,
                                                _: String ->

            when (rc) {
                OK -> {
                    Toast.makeText(this, getString(R.string.bio_init), LENGTH_SHORT).show()

                    App.DevFamily = App.BioManager!!.deviceFamily
                    App.DevType = App.BioManager!!.deviceType

                    val intent = when (App.DevFamily) {
                        DeviceFamily.CredenceTAB ->
                            Intent(this, com.credenceid.sample.epassport.ctab.MRTDActivity::class.java)
                        DeviceFamily.CredenceTwo ->
                            Intent(this, com.credenceid.sample.epassport.ctwo.MRZActivity::class.java)
                        DeviceFamily.CredenceThree ->
                            Intent(this, com.credenceid.sample.epassport.eco.mrtd.MRTDActivity::class.java)
                        DeviceFamily.CredenceECO ->
                            Intent(this, com.credenceid.sample.epassport.eco.mrtd.MRTDActivity::class.java)
                        else -> return@initializeBiometrics
                    }
                    intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
                    startActivity(intent)
                    this.finish()

                }
                INTERMEDIATE -> {
                    /* This code is never returned for this API. */
                }
                FAIL -> Toast.makeText(this, getString(R.string.bio_ini_fail), LENGTH_LONG).show()
            }
        }
    }


    // Check permissions at runtime
    fun checkPermissions() {
        if (isPermissionsGranted() != PackageManager.PERMISSION_GRANTED) {
            ActivityCompat.requestPermissions(this@LaunchActivity, list.toTypedArray(), PERMISSION_REQUEST_CODE)
        } else {
            this.initBiometrics()
        }
    }

    val list = if (android.os.Build.VERSION.SDK_INT <= 30) {
        listOf(
            Manifest.permission.CAMERA,
        )
    } else {
        listOf(
            Manifest.permission.CAMERA,
        )
    }


    // Check permissions status
    private fun isPermissionsGranted(): Int {
        var counter = 0;
        for (permission in list) {
            counter += ContextCompat.checkSelfPermission(this, permission)
        }
        return counter
    }
}
